docs/ai/shade-agents/introduction.md

```diff
@@ -0,0 +1,94 @@
+
+These agents can autonomously sign transactions across any chain, interact with AI models and external data sources, manage assets, and perform privacy-preserving, verifiable computations, offering the flexibility and performance of Web2 with the verifiability of Web3.
+
+:::info
+Shade Agents power [Agentic Protocols](./examples.md#agentic-protocols): a new type of decentralized application designed to be autonomous, proactive, and intelligent.
+:::
+
+---
+
+## How do Shade Agents work?
+
+Shade Agents consist of two main components: a `worker agent` and an `agent smart contract`.
+
+![Shade Agent Architecture](/docs/assets/shade-agents/shade-agent-stack-diagram.png)
+
+When a worker agent is booted up in a TEE, the `TEE's hardware-based entropy` generates a random private key and account. This private key is exclusively used to call the agent contract, not for storing funds.
+
+The worker agent calls the `register_worker` function on the agent contract, providing two pieces of information generated inside the TEE:
+- A `remote attestation quote` (which proves it is running inside a genuine TEE).
+- The Docker image's SHA256 `code hash` (to prove that the expected code is running).
+
+If the attestation quote is valid and the code hash matches the expected code hash of the worker agent (defined during agent contract deployment), the worker agent's account is approved as a valid worker agent.
+
+Once registered, the worker agent can call the `sign_tx` function on the agent contract, enabling it to sign transactions on behalf of the Shade Agent. The Shade Agent utilizes [chain signatures](../../chain-abstraction/chain-signatures.md) for decentralized key management, allowing it to hold assets and sign transactions on nearly any chain.
+
+Using the correct Docker image, `anyone` can deploy a worker agent to a TEE, register as a valid worker, and access the sign_tx function to control the same Shade Agent accounts. This means the accounts are never lost, and the service is `always live`. Worker agents are designed to be `stateless` to facilitate this functionality.
+
+<details>
+
+<summary>What is a TEE?</summary>
+
+A trusted execution environment is a secure area of a CPU that runs code in an isolated and protected way. This means we know the expected code is running and its execution is not exposed outside of the enclave. TEEs produce attestations to prove that the code is running within a TEE and that it's running the specified code.
+
+</details>
+
+---
+
+## Agent Contract Functions
+
+The Agent Contract has three main functions:
+
+#### Approve Code Hash
+
+After contract deployment, a `code hash` is set to ensure that only worker agents running the correct worker code can register.
+
+Developers or protocols may need to modify the code running inside worker agents. When the contract is first deployed, it's initialized with an `owner` account Id. Only the owner can call the approve code hash function to update the worker agent code.
```

Comment:
```suggestion
Developers or protocols may need to modify the agent's code. Because of this, when the contract is initialized a `owner` is set, which is the only account allowed to approve updates to the stored `code hash`.
```